### PR TITLE
Add coordinate janicart pda

### PR DESCRIPTION
**Pull request by PIerreLouisH**
## What Does This PR Do
It add the coordinate of the janicart upon the janitor PDA cartride (=Custodial Locator)

## Why It's Good For The Game
It allow the janitor to finally retrieve their janicart after those damn <del>voxs</del> thieves stole it. 
Good luck finding your keys however!

## Images of changes
image

## Testing
Looked upon the janitor PDA and changed coordinate upon different Z-levels

## Changelog
:cl:
add: Added janicart coordinates upon the janitor PDA
/:cl:

### CODE REVIEW

**Comment by S34NW** [comment]
PR frozen during TGUI upgrade project, announcement detailing this can be found here:
https://discord.com/channels/145533722026967040/499000353312735233/1181289070752710657

Feel free to ping me on discord if you want to help out with that while you wait for this to be unfrozen :)

**Comment by PIerreLouisH** [comment]
TGUI unfrozen, hence I can reopen this old PR. 

I updated my branch with master and tested without any issues.
image

File: code/modules/pda/cart_apps.dm [hunk 1]
```diff
@@ -449,8 +449,19 @@
...
 			CartData[++CartData.len] = list("x" = bl.x, "y" = bl.y, "dir" = uppertext(dir2text(direction)), "volume" = B.reagents.total_volume, "max_volume" = B.reagents.maximum_volume)
 
+	var/list/JaniCartData = list()
+	for(var/obj/vehicle/janicart/J in GLOB.janitorial_equipment)
+		var/turf/jl = get_turf(J)
+		if(jl)
+			if(jl.z != cl.z)
+				continue
+			var/direction = get_dir(pda,J)
+			JaniCartData[++JaniCartData.len] = list("x" = jl.x, "y" = jl.y, "dir" = uppertext(dir2text(direction)))
```
Comment (S34NW) [inline code/modules/pda/cart_apps.dm:459]:
```suggestion
			var/direction_from_user = get_dir(pda, janicart)
			JaniCartData[++JaniCartData.len] = list("x" = janicart_loc.x, "y" = janicart_loc.y, "direction_from_user" = uppertext(dir2text(direction_from_user)))
```

File: code/modules/vehicle/janivehicle.dm [hunk 2]
```diff
@@ -11,7 +11,13 @@
 	/// Does it clean the tile under it?
 	var/floorbuffer = FALSE
 
+/obj/vehicle/janicart/Initialize(mapload)
+	. = ..()
+	GLOB.janitorial_equipment += src
+
```
Comment (S34NW) [inline code/modules/vehicle/janivehicle.dm:17]:
```suggestion
```

File: tgui/packages/tgui/interfaces/pda/pda_janitor.js [hunk 3]
```diff
@@ -48,6 +48,15 @@ export const pda_janitor = (props, context) => {
           ))}
         </LabeledList.Item>
       )}
+      {janicarts && (
+        <LabeledList.Item label="Janicart Locations">
+          {janicarts.map((j) => (
+            <Box key={j}>
+              {j.x},{j.y} ({j.dir})
```
Comment (S34NW) [inline tgui/packages/tgui/interfaces/pda/pda_janitor.js:55]:
```suggestion
              {janicart.x},{janicart.y} ({janicart.direction_from_user})
```

File: tgui/packages/tgui/interfaces/pda/pda_janitor.js [hunk 4]
```diff
@@ -48,6 +48,15 @@ export const pda_janitor = (props, context) => {
           ))}
         </LabeledList.Item>
       )}
+      {janicarts && (
+        <LabeledList.Item label="Janicart Locations">
+          {janicarts.map((j) => (
```
Comment (S34NW) [inline tgui/packages/tgui/interfaces/pda/pda_janitor.js:53]:
```suggestion
          {janicarts.map((janicart) => (
```

File: tgui/packages/tgui/interfaces/pda/pda_janitor.js [hunk 5]
```diff
@@ -48,6 +48,15 @@ export const pda_janitor = (props, context) => {
...
+            <Box key={j}>
```
Comment (S34NW) [inline tgui/packages/tgui/interfaces/pda/pda_janitor.js:54]:
```suggestion
            <Box key={janicart}>
```

File: code/modules/pda/cart_apps.dm [hunk 6]
```diff
@@ -449,8 +449,19 @@
 			var/direction = get_dir(pda,B)
 			CartData[++CartData.len] = list("x" = bl.x, "y" = bl.y, "dir" = uppertext(dir2text(direction)), "volume" = B.reagents.total_volume, "max_volume" = B.reagents.maximum_volume)
 
+	var/list/JaniCartData = list()
+	for(var/obj/vehicle/janicart/J in GLOB.janitorial_equipment)
```
Comment (S34NW) [inline code/modules/pda/cart_apps.dm:453]:
```suggestion
	for(var/obj/vehicle/janicart/janicart in GLOB.janitorial_equipment)
```

File: code/modules/pda/cart_apps.dm [hunk 7]
```diff
@@ -449,8 +449,19 @@
...
+		var/turf/jl = get_turf(J)
```
Comment (S34NW) [inline code/modules/pda/cart_apps.dm:454]:
```suggestion
		var/turf/janicart_loc = get_turf(janicart )
```

File: code/modules/pda/cart_apps.dm [hunk 8]
```diff
@@ -449,8 +449,19 @@
...
+		if(jl)
```
Comment (S34NW) [inline code/modules/pda/cart_apps.dm:455]:
```suggestion
		if(janicart_loc)
```

File: code/modules/pda/cart_apps.dm [hunk 9]
```diff
@@ -449,8 +449,19 @@
...
+			if(jl.z != cl.z)
```
Comment (S34NW) [inline code/modules/pda/cart_apps.dm:456]:
```suggestion
			if(janicart_loc.z != cl.z)
```

**Review by S34NW — COMMENTED** [review]
Functionality is fine, some readability tweaks before merge. Also please rebuild TGUI.

**Review by S34NW — APPROVED** [review]

**Pull request merged**